UnityProject/Assets/Scripts/Core/ITrackableType.cs

```diff
@@ -9,6 +9,25 @@ namespace Core
 	public static class ComponentsTracker<T>
 	{
 		public static HashSet<T> Instances { get; } = new HashSet<T>();
+		private static Dictionary<GameObject, T> instanceLookup = new Dictionary<GameObject, T>();
```

Comment:
there's already a system for this GetComponentCustom, though it does need a CommonComponents on the object

Reply:
CommonComponents still does a GetComponent. This bypasses GetComponent entirely.

Reply:
well You could make it add itself to common components though?

Reply:
Adding UniversalPhysics to CommonComponents while trying to get UniversalPhysics that has a reference to CommonComponnets?
nu uh